### PR TITLE
Comments can be viewed from Post Details

### PR DESCRIPTION
**Pull request by matt-rich-anderson**
Referencing ticket #29 

#### Changes Made
1. Added file `Comments/Comments.js` to the `components` directory.
2. Modified file `authManager.js` to include get comments fetch call.
3. Modified file `ApplicationViews.js` to include get comments page routing.
4.  Modified file `PostDetails.js` to include get "View Comments" button.
5.  Added the `Comment.cs` class to `Models` directory.
6.  Modified the `Post.js` class to contain the Comments property.
7.  Modified files `PostRepository.cs` and `IPostRepository.cs` to contain query methods.
8. Modified file `PostController.cs` to handle Controllers HTTP requests.

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout ma-ViewComments
    ```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
4. Test app functionality.
    As a logged-in user:
    >Click Posts in the navbar, click on the title of a particular post

    >Click the "View Comments" button - you should see comments listed (recent first) followed by username and create date
    
    >Click "Return to Post" and you should be taken back to post details
5. View code file.
    > Confirm file modifications are present as indicated above.
    
   > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

File: Tabloid/Controllers/PostController.cs [hunk 1]
```diff
@@ -24,6 +24,13 @@ public IActionResult Get()
             return Ok(_postRepo.GetAllPublishedPosts());
         }
 
+        [HttpGet("comments/{id}")]
+        public IActionResult GetWithComments(int id)
+        {
+            var post = _postRepo.GetPostByIdWithComments(id);
+            return Ok(post);
+        }
+
```
Comment (SchmidtFred) [inline Tabloid/Controllers/PostController.cs:36]: This is returning nothing if there are no comments on the post.

File: Tabloid/client/src/components/Comments/Comments.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <>
+      <h2>{postWithComments.title}</h2>
+      <div>
+        <ListGroup>
+          {postWithComments.comments?.map((c) => {
+            return (
+              <ListGroupItem>
+                {c.subject} {c.content}{" "}
+                <b>
```
Comment (SchmidtFred) [inline Tabloid/client/src/components/Comments/Comments.js]: Each item inside of a map in React requires the key prop.

**Review by SchmidtFred — CHANGES_REQUESTED** [review]
Check comments for requested changes.

File: Tabloid/Controllers/PostController.cs (hunk 1 above)
Comment (matt-rich-anderson) [reply Tabloid/Controllers/PostController.cs:36]: Server-side is now returning what it should

File: Tabloid/client/src/components/Comments/Comments.js (hunk 2 above)
Comment (matt-rich-anderson) [reply Tabloid/client/src/components/Comments/Comments.js]: key prop has been added

**Review by SchmidtFred — APPROVED** [review]

**Review by IsaacWeiser — APPROVED** [review]

**Pull request merged**